test/framework/provider.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"time"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Wait for Provider to be successfully installed.
+func WaitForProviderToBeSuccessfullyInstalled(ctx context.Context, c client.Client) error {
```

Comment:
Could we make this `WaitForAllProvidersInstalled` to more clearly indicate the functionality? Also, might be nice to expose timeout and interval in call.